### PR TITLE
issue: .eml/.msg Attachments

### PR DESCRIPTION
**Pull request by JediKev**
This addresses an issue where `.eml` and `.msg` files on incoming mails are being dropped. This is due the the mail fetcher that tries to process `.eml`/`.msg` files and adds them as thread entries rather than adding them as attachments. This adds a new section that utilizes a new method to fetch the body of `.eml`/`.msg` files, fetches the subjects of the `.eml`/`.msg` files as the attachment names, and creates attachments. This preserves the `.eml` and `.msg` files and adds them to the pertinent thread entries as attachments.

### CODE REVIEW

File: include/class.mailfetch.php [hunk 1]
```diff
@@ -675,6 +693,26 @@ function createTicket($mid) {
...
                     }
+                    break;
+                // Parse attached email message
+                case 'message/rfc822':
+                    try {
+                        // Fetch the header of attached mime message.
+                        $body = $this->fetchBody($mid, $info['index'].'.0',
+                            $info['encoding']);
+                        // Add fake body to make the parser happy
+                        if ($body)
```
Comment (protich) [inline include/class.mailfetch.php]: Wonder is this was meant to be `if(!$body)`

Comment (JediKev) [reply include/class.mailfetch.php]: I already have this in my notes to discuss with you tomorrow 😂

**Comment by protich** [comment]
@JediKev - I think we need equivalent implementation for mail parse (piping).

**Comment by JediKev** [comment]
@protich Will look into it.

**Comment by protich** [comment]
Check stashed changes before looking into it... I don't remember porting the changes to `class.mailparse.php` but it has been many moons ;)

**Comment by JediKev** [comment]
@protich Aye aye cap’n!

**Comment by FG-Lars** [comment]
This issue is back in v1.14.1 and this file only screws up that version.

Anybody up for fixing that? ^^,

**Comment by JediKev** [comment]
@FG-Lars 

Steps to replicate? An example .eml attachment?

Cheers.

**Comment by FG-Lars** [comment]
Never mind.

Earlier attachments have been .msg and this one was .eml

Did not have .eml in allowed list.

**Comment by JediKev** [comment]
@FG-Lars 

lol awesome! Worried me for a minute because I know we haven't touched that since 😅

Cheers.

**Comment by FG-Lars** [comment]
Sorry for the scare ^^,

**Comment by NCaur** [comment]
Hi, though I am able to rectify the problem; I would like to know why .eml files were getting dropped; specially when sent via server but not if sent by client such as Outlook.
Thank you.

**Pull request merged**